README.md

```diff
@@ -0,0 +1,134 @@
+
+The library provides the following functionalities:
+* Implementation of the FFT in the complex realm with a 64-bit representation of complex coefficients.
+* Fast multiplication of polynomials in the ring Z[X]/(X^N+1) or R[X]/(X^N+1) with up to 52 bits of precision.
+
+
+### Optimisations
+
+The library auto-detects support for AVX, AVX2, FMA, AVX512F instructions at runtime and automatically switches to the 
+most efficient implementation, without requiring to be compiled on the spot on the target machine. This allows to distribute 
+not only source but also binary distributions of the library via standard package managers (e.g. apt-install) 
+
+
+## Dependencies 
+
+Spqlios-library is a C library that can be compiled with a standard C compiler, and depends only on libc and libm. The API 
+interface can be used in a regular C code, and any other language via classical foreign APIs. 
+
+The unittests and integration tests are in an optional part of the code, and are written in C++.  These tests rely on 
+[```benchmark```](https://github.com/google/benchmark), and ```gtest``` libraries, and therefore require a C++17 compiler.
+
+Currently, the project has been tested with the gcc,g++ >= 11.3.0 compiler under Linux (x86_64). In the future, we plan to
+extend the compatibility to other compilers, platforms and operating systems. 
+
+
+## Installation
+
+To build the library with the default options, create a ```build``` folder in the top level directory and run ```make``` from inside it. This assumes that the standard tool cmake is already installed on the system, and an up-to-date c++ compiler (i.e. g++ >=11.3.0) as well.
+
+It will compile the shared library in optimized mode, and ```make install``` install it to the desired prefix folder (by default ```/usr/local/lib```).
+
+If you want to choose additional compile options (i.e. other installation folder, debug mode, tests), you need to run cmake manually and pass the desired options:
+```
+mkdir build
+cd build
+cmake ../src -CMAKE_INSTALL_PREFIX=/usr/
+make
+```
+The available options are the following:
+
+| Variable Name          | values           | 
+|------------------------|-------|
+| CMAKE_INSTALL_PREFIX   | */usr/local* installation folder (libs go in lib/ and headers in include/) | 
+| WARNING_PARANOID | All warnings are shown and treated as errors. Off by default |
+| ENABLE_TESTING | Compiles unit tests and integration tests |
+
+
+
+## Usage
+
+### Layouts for complex FFT
+
+The complex FFT handles polynomials with complex coefficients `C[X] mod X^M-i`, each one represented by two consecutive doubles `(real,imag)`. Note that a real polynomial $$\sum_{j=0}^{N-1} p_j\cdot X^j \mathtt{ mod } X^N+1$$ corresponds to the complex polynomial of half degree `M=N/2`: 
+$$\sum_{j=0}^{M-1} (p_{j} + i.p_{j+M}) \cdot X^j \mathtt{ mod } X^M-i$$
```

Comment:
add:
In both cases, the FFT of the polynomial is the vector of its (complex) evaluations on the $M$ roots of $X^M-i$.

Reply:
added